src/odemis/acq/test/stream_test.py

```diff
@@ -2243,6 +2210,191 @@ def test_acq_cl_leech(self):
         self.assertGreater(len(pcmd), 500 / 10)
 
 
+class SPARC2TestCaseStageWrapper(unittest.TestCase):
+    """
+    This test case is specifically targeting the use of a stage wrapper to
+    enable stage scanning with the SEM sample stage.
+    """
+    # The hardware is very similar to the SPARCv1, so just check special behaviour
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            testing.start_backend(SPARC2_4SPEC_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Find CCD & SEM components
+        cls.microscope = model.getMicroscope()
+        cls.ccd = model.getComponent(role="sp-ccd1")
+        cls.spec = model.getComponent(role="spectrometer")
+        cls.ebeam = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.scan_stage = model.getComponent(role="scan-stage")
+        cls.sem_stage = model.getComponent(role="sem-stage")
+        cls.ebic = model.getComponent(role="ebic-detector")
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        testing.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    def test_scan_stage_wrapper(self):
+        """
+        Simple test case to check if the scan stage wrapper works like expected.
+        This wrapper setup is needed when there is no dedicated (physical) scan stage
+        """
+        # Create the SEM and Spectrum streams
+        sems = stream.SEMStream("test sem cl", self.sed, self.sed.data, self.ebeam)
+        specs_sstage = stream.SpectrumSettingsStream("test spec",
+                                              self.spec,
+                                              self.spec.data,
+                                              self.ebeam,
+                                              sstage=self.scan_stage)
+        sps = stream.SEMSpectrumMDStream("test sem-spec", [sems, specs_sstage])
+
+        # set stage scanning to true
+        specs_sstage.useScanStage.value = True
+
+        # Keep ROI minimal to keep the acquisition time as low as possible with a low repetition
+        #     roi ~ 20um x 20um
+        #     rep 5, 5
+        specs_sstage.roi.value = (0.492, 0.492, 0.508, 0.508)
+        specs_sstage.repetition.value = (5, 5)
```

Comment:
Could you change it so that it's not square. Having it a rectangle means it's possible to detect if X/Y are swapped, which is a very common mistake.
You can use for example 4x6 (with the equivalent aspect ratio for the roi.